crates/searchers/openapi.yml

```diff
@@ -0,0 +1,605 @@
+        takerToken:
+          $ref: "#/components/schemas/Token"
+        makerAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        takerAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        takerTokenFeeAmount:
+          $ref: "#/components/schemas/TokenAmount"
+
+    LiquidityParameters:
+      oneOf:
+        - $ref: "#/components/schemas/ConstantProductPool"
+        - $ref: "#/components/schemas/WeightedProductPool"
+        - $ref: "#/components/schemas/StablePool"
+        - $ref: "#/components/schemas/ConcentratedLiquidityPool"
+        - $ref: "#/components/schemas/ForeignLimitOrder"
+
+    Liquidity:
+      description: |
+        On-chain liquidity that can be used in a solution. This liquidity is
+        provided to facilitate onboarding new solvers. Additional liquidy that
+        is not included in this set may still be used in solutions.
+      allOf:
+        - $ref: "#/components/schemas/LiquidityParameters"
+        - type: object
+          properties:
+            id:
+              description: |
+                An opaque ID used for uniquely identifying the liquidity within
+                a single auction (note that they are **not** guaranteed to be
+                unique across auctions). This ID is used in the solution for
+                matching interactions with the executed liquidity.
+              type: string
+            address:
+              $ref: "#/components/schemas/Address"
+            gasEstimate:
+              type: integer
+              description: |
+                A rough approximation of gas units required to
+                use this liquidity on-chain.
+
+    Auction:
+      description: |
+        The abstract auction to be solved by the searcher.
+      type: object
+      required:
+        - tokens
+        - orders
+        - liquidity
+        - gasPrice
+      properties:
+        id:
+          type: integer
+          description: |
+            The identifier for the auction. Will be set to `null` for requests
+            that are not part of an auction (when quoting token prices for
+            example).
+        tokens:
+          type: object
+          additionalProperties:
+            $ref: "#/components/schemas/TokenInfo"
+          description: |
+            A map of token addresses to token information.
+        orders:
+          type: array
+          items:
+            $ref: "#/components/schemas/Order"
+          description: |
+            The solvable orders included in the auction.
+        liquidity:
+          type: array
+          items:
+            $ref: "#/components/schemas/Liquidity"
+          description: |
+            On-chain liquidity that can be used by the solution.
+        gasPrice:
+          description: |
+            The current estimated gas price. This is the price that will be used
+            for solution scoring.
+          $ref: "#/components/schemas/TokenAmount"
```

Comment:
This might be the good moment to do this: https://github.com/cowprotocol/services/pull/794#pullrequestreview-1184077784

Reply:
🤔 - this would require changing our objective value computation, which I am not a fan of (basically, this is the value that we multiply `eth_estimateGas` result with.

I'm going to rename to `effecitiveGasPrice` as that is:
- An established term for what we want to express here (the gas price we actually pay)
- Is "compatible" with EIP-1559 (in that it is a concept introduced in that EIP).

Reply:
> this would require changing our objective value computation

But you can always call `GasPrice1559::efective_gas_price()` when you need it?

I was pretty convinced that we should pass around `GasPrice1559` as a `gas_price` everywhere in our code, and then just convert it to whatever we need at the call site:
1. Into `ethcontract::GasPrice` for simulation/execution
2. into f64 for computation of costs - obj values whatever

For example, we send gas price as part of the `BatchAuctionModel` here: https://github.com/cowprotocol/services/blob/b4a6c001329abe6282751ad8d243972a15206eeb/crates/solver/src/solver/http_solver.rs#L206 which is an outdated value and should be changed to `<max_fee_per_gas, max_priority_fee_per_gas>` IMO (although I am not sure what it is used for in this context, maybe `base_fee` is also needed).
But again, I might be missing something